plugins/exec/heap.ml

```diff
@@ -0,0 +1,56 @@
+open Extends
+open Coqnum
+open Coqunit
+
+let keys = Stream.from (fun x -> Some x)
```

Comment:
Hmm, why don't you simply use a reference containing the last used integer?

Reply:
I find calling `Stream.next` more elegant and straightforward than manipulating a reference myself. That being said, I can implement the solution you propose if you really prefer.

Reply:
I am still unconvinced by the advantage of the stream-based approach: not only this solution seems operationally equivalent to a simple reference but it has a computational cost, as you have to pay an indirect call and an extra unboxing comparing. I may be missing something but if I do not, I would suggest a simpler, more idiomatic, reference-based approach.

Reply:
Let’s do that then. (: